sktime/classifiers/proximity.py

```diff
@@ -0,0 +1,1084 @@
+            distribution = [0] * num_unique_class_labels
+            distributions.append(distribution)
+            # invert distances (as larger distance should be less likely predicted)
+            for exemplar_index in range(0, num_exemplars):
+                distance = distances[instance_index][exemplar_index]
+                exemplar_class_label = self.exemplar_class_labels[exemplar_index]
+                distribution[exemplar_class_label] -= distance
+            max_distance = -np.min(distribution)
+            for exemplar_index in range(0, num_exemplars - 1):
+                distribution[exemplar_index] += max_distance
+        # normalise inverted distances to a probability
+        distributions = np.array(distributions)
+        normalize(distributions, copy = False, norm = 'l1')
+        return distributions
+
+    def _find_distance(self, instance_a, instance_b, should_check_data = True):
+        '''
+        find distance between two instances using distance measure + distance measure parameters
+        ----
+        Parameters
+        ----
+        instance_a : panda dataframe
+            instance of the dataset
+        instance_a : panda dataframe
+            another instance of the dataset
+        should_check_data : boolean
+            whether to verify the dataset (e.g. dimensions, etc)
+        ----
+        Returns
+        ----
+        distance : float
+            value indicating how similar the two instances are
+        '''
+        if should_check_data:
+            if not isinstance(instance_a, Series):
+                raise ValueError("instance not a panda series")
+            if not isinstance(instance_b, Series):
+                raise ValueError("instance not a panda series")
+        # flatten both instances and transpose for cython parameter format
+        instance_a = tabularise(instance_a, return_array = True)
+        instance_b = tabularise(instance_b, return_array = True)
+        instance_a = np.transpose(instance_a)
+        instance_b = np.transpose(instance_b)
+        # find distance
+        return self.distance_measure(instance_a, instance_b, **self.distance_measure_param_perm)
+
+
+class ProximityTree(Classifier):  # todd rename split to stump
+
+    '''
+    proximity tree classifier using proximity stumps at each tree node to split data into branches.
+    ----
+    Parameters
+    ----
+    pick_exemplars_method : callable
+        Method to pick exemplars from a set of instances and class labels
+    param_pool : list of dicts
+        a list of dictionaries containing a distance measure and corresponding parameter sources (distribution or
+        predefined value)
+    gain_method : callable
+        a method to calculate the gain of this split / stump
+    label_encoder : LabelEncoder
+        a label encoder, can be pre-populated
+    rand : numpy RandomState
+        a random state for sampling random numbers
+    r : int
```

Comment:
Maybe rename r to n_stumps?

Reply:
It's called `r` in the paper and Java code so I simply mimicked them. I think `n_stumps` is better, what do you think @jasonlines @TonyBagnall ?